### ISSUE TITLE
Nuxt error pinia after command yarn start

### ISSUE THREAD

**Issue by alessioraggioli** [nuxt/nuxt#19313]
### Environment

prod

### Reproduction

Hello,
seems like pinia is working only in dev mode.
After the command yarn start, I receive a compilation error

Cannot find module 'C:\Users\Alessio\Documents\nuxt3-gfl-base-template\.output\server\node_modules\pinia\dist\pinia' imported from C:\Users\Alessio\Documents\nuxt3-gfl-base-template\.output\server\chunks\config.mjs

### Describe the bug

Cannot find module 'C:\Users\Alessio\Documents\nuxt3-gfl-base-template\.output\server\node_modules\pinia\dist\pinia' imported from C:\Users\Alessio\Documents\nuxt3-gfl-base-template\.output\server\chunks\config.mjs

### Additional context

{
  "private": true,
  "overrides": {
    "vue": "latest"
  },
  "scripts": {
    "dev": "nuxt dev --port 4100",
    "build": "nuxt build",
    "start": "nuxt start",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "postinstall": "nuxt prepare"
  },
  "devDependencies": {
    "@nuxtjs/apollo": "^5.0.0-alpha.5",
    "@nuxtjs/google-fonts": "^3.0.0-0",
    "@nuxtjs/moment": "^1.6.1",
    "nuxt": "^3.2.1",
    "sass": "^1.55.0",
    "sass-loader": "10",
    "vite-plugin-fonts": "^0.6.0"
  },
  "dependencies": {
    "@mdi/font": "^7.0.96",
    "@pinia/nuxt": "^0.4.7",
    "fs.promises": "^0.1.2",
    "pinia": "^2.0.32",
    "pinia-plugin-persistedstate": "^2.3.0",
    "vuetify": "^3.0.0"
  }
}

**Comment by Baroshem** [comment]
Hey, 

I believe this is a bug from pinia / pinia module for nuxt rather than Nuxt itself.

Have you tried reporting this bug in the pinia repo? https://github.com/vuejs/pinia/tree/v2/packages/nuxt

**Comment by alessioraggioli** [comment]
Thanks @Baroshem , not yet. I'm opening the issue to them as well.

**Comment by Baroshem** [comment]
Good, make sure to link the issue here so that we can keep track of it :)

**Comment by alessioraggioli** [comment]
https://github.com/vuejs/pinia/issues/2032

**Comment by alessioraggioli** [comment]
@Baroshem maybe I've found the real problem.

I have a problem loading files in my server/api/config.js
This is the file: 

import { readFile } from 'fs/promises'

async function readJsonFile(path) {
  const file = await readFile(path, 'utf8')
  return JSON.parse(file)
}

export default defineEventHandler(async () => {
  try {
    const env = process.env.ENV
    
    if (env) { //prova a lanciare l'error da qui
      const envPath = `./data/${env}.json`
      const configuration = await readJsonFile(envPath)
      console.log(configuration)
      console.log('///////INIT ENVIRONMENT: ' + env)
      return configuration
    }
    else{
      return false
    }
  } catch (e) {
    console.log('///////ERROR LOADING ENVIRONMENT: ' + e.message)
    return false
  }
})

I've removed the part where I was loading the pinia store, but the problems seems that the building proceedment doesn't care about my data folder.

ERROR LOADING ENVIRONMENT: ENOENT: no such file or directory, open 'C:\Users\Alessio\Documents\test-n3\.output\data\preprod.json'

This error doesn't show in dev mode, just after yarn build and yarn start.
Any tips?

Thanks
